### PR TITLE
fixed V! stack pane stuck

### PR DESCRIPTION
**Pull request by Vane11ope**
how to reproduce the bug:

1. go to V!
2. push tab until you get to stack pane
3. keep pushing j and you will see it gets stuck

### CODE REVIEW

File: libr/core/panels.c [hunk 1]
```diff
@@ -119,6 +119,79 @@ static const char **menus_sub[] = {
 	NULL
 };
 
+static char *str_crop(const char *str, ut32 x, ut32 y, ut32 x2, ut32 y2) {
```
Comment (radare) [inline libr/core/panels.c]: Why not use r_str_crop?

File: libr/core/panels.c [hunk 2]
```diff
@@ -119,6 +119,79 @@ static const char **menus_sub[] = {
...
+	char *r, *r_end, *ret;
+	const char *s;
+	size_t r_len, str_len = 0, nr_of_lines = 0;
+	ut32 currentHeight = 0, currentWidth = 0;
+	if (x2 < 1 || y2 < 1 || !str) {
+		return strdup ("");
```
Comment (radare) [inline libr/core/panels.c]: Why not return null

File: libr/core/panels.c (hunk 1 above)
Comment (Vane11ope) [reply libr/core/panels.c]: no it should be able to use ansi crop which has a bug inside that works properly for VV which i dont know why.

if i fix it the way that should be fixed then VV gets glitches as you once told me

**Comment by radare** [comment]
hehe, this should be changed, this may save us to do an unnecessary malloc+print+free for nothing

**Comment by radare** [comment]
uhm, will be good to find whats the problem in current r_str_crop, how can i reproduce the problem?

**Comment by Vane11ope** [comment]
okay so go to r_ansi_crop or something in str.c

then you know the api essentially should just rip the line of string off if it isnt in range x1-x2 and y1-y2 right

but if the string has ansi codes in it, then it keeps the ansi parts and just only rips off the char parts so there're tons of unnecessary ansi garbages like ghost.

you'll easily see that by creating log files that keep both the string before API applied and after

**Comment by radare** [comment]
Thanks! looks good

**Pull request merged**